WalletWasabi.Fluent/ViewModels/Wallets/WalletViewModel.cs

```diff
@@ -33,19 +36,138 @@ public partial class WalletViewModel : WalletViewModelBase
 	[AutoNotify(SetterModifier = AccessModifier.Private)] private bool _isTransactionHistoryEmpty;
 	[AutoNotify(SetterModifier = AccessModifier.Private)] private bool _isSendButtonVisible;
 
-	protected WalletViewModel(Wallet wallet) : base(wallet)
+	[AutoNotify(SetterModifier = AccessModifier.Protected)]
+	private bool _isLoading;
+
+	[AutoNotify(SetterModifier = AccessModifier.Protected)]
+	private bool _isCoinJoining;
+
+	public WalletState WalletState => Wallet.State;
+
+	private string _title;
+
+	public Wallet Wallet { get; }
+
+	public string WalletName => Wallet.WalletName;
+
+	public bool IsLoggedIn => Wallet.IsLoggedIn;
+
+	public bool PreferPsbtWorkflow => Wallet.KeyManager.PreferPsbtWorkflow;
+
+	public override string ToString() => WalletName;
+
+	public int CompareTo(WalletViewModel? other)
+	{
+		if (other is null)
+		{
+			return -1;
+		}
+
+		var result = other.IsLoggedIn.CompareTo(IsLoggedIn);
+
+		if (result == 0)
+		{
+			result = string.Compare(Title, other.Title, StringComparison.Ordinal);
+		}
+
+		return result;
+	}
+
+	private WalletViewModel(WalletPageViewModel parent)
+	{
+		_parent = parent;
+		Wallet = parent.Wallet;
+	}
+
+	private bool _isInitialized;
+
+	public UiTriggers UiTriggers { get; private set; }
+
+	public CoinJoinSettingsViewModel CoinJoinSettings { get; private set; }
+
+	public bool IsWatchOnly => Wallet.KeyManager.IsWatchOnly;
+
+	[AutoNotify(SetterModifier = AccessModifier.Private)]
+	private IObservable<bool> _isMusicBoxVisible;
+
+	internal CoinJoinStateViewModel CoinJoinStateViewModel { get; private set; }
+
+	public WalletSettingsViewModel Settings { get; private set; }
+
+	public ICommand SendCommand { get; private set; }
+
+	public ICommand? BroadcastPsbtCommand { get; set; }
+
+	public ICommand ReceiveCommand { get; private set; }
+
+	public ICommand WalletInfoCommand { get; private set; }
+
+	public ICommand WalletSettingsCommand { get; private set; }
+
+	public ICommand WalletStatsCommand { get; private set; }
+
+	public ICommand WalletCoinsCommand { get; private set; }
+
+	public ICommand CoinJoinSettingsCommand { get; private set; }
+
+	private CompositeDisposable Disposables { get; set; }
+
+	[AutoNotify(SetterModifier = AccessModifier.Private)] private HistoryViewModel _history;
+	[AutoNotify(SetterModifier = AccessModifier.Private)] private IEnumerable<ActivatableViewModel> _tiles;
+
+	public void NavigateAndHighlight(uint256 txid)
+	{
+		Navigate().To(this, NavigationMode.Clear);
+
+		RxApp.MainThreadScheduler.Schedule(async () =>
+		{
+			await Task.Delay(500);
```

Comment:
What is it for?

Reply:
I assume it waits for screen to load. This code exists in current master so it's not related to this PR.